### ISSUE TITLE
NullReferenceException on document.getElementById('')

### ISSUE THREAD

**Issue by cihanuygun** [Rionmonster/Glyphfriend#45]
While typing document.getElementById() when you start to type parameter value inside getElementById method, Glyphfriend Extension causes Visual Studio to crash. The details are below:

Extension Version: v1.4.2 (Updated on 15.04.2016)

The stack of the exception:

> Application: devenv.exe
Framework Version: v4.0.30319
Description: The process was terminated due to an unhandled exception.
Exception Info: System.NullReferenceException
   at Glyphfriend.GlyphCompletionProviders.FontAwesomeGlyphCompletionProvider.GetCompletionGlyph(System.String, System.Uri, Microsoft.CSS.Editor.Completion.CssNameType)
   at Microsoft.CSS.Editor.Document.Cache.TreeCachedNameCollector.CollectClassName(Microsoft.CSS.Core.Parser.ParseItem)
   at Microsoft.CSS.Editor.Document.Cache.TreeCachedNameCollector.Visit(Microsoft.CSS.Core.Parser.ParseItem)
   at Microsoft.CSS.Core.Parser.ComplexItem.Accept(Microsoft.CSS.Core.Parser.ICssSimpleTreeVisitor)
   at Microsoft.CSS.Core.Parser.ComplexItem.Accept(Microsoft.CSS.Core.Parser.ICssSimpleTreeVisitor)
   at Microsoft.CSS.Core.Parser.ComplexItem.Accept(Microsoft.CSS.Core.Parser.ICssSimpleTreeVisitor)
   at Microsoft.CSS.Core.Parser.ComplexItem.Accept(Microsoft.CSS.Core.Parser.ICssSimpleTreeVisitor)
   at Microsoft.CSS.Core.Parser.ComplexItem.Accept(Microsoft.CSS.Core.Parser.ICssSimpleTreeVisitor)
   at Microsoft.VisualStudio.Html.Package.Project.CSS.CssProjectItemData.GetNames(Microsoft.CSS.Core.TreeItems.StyleSheet, System.Threading.CancellationToken)
   at Microsoft.VisualStudio.Html.Package.Project.CSS.CssProjectItemData+<>c__DisplayClass43_2.<StartUpdateClassNameCache>b__1()
   at System.Threading.Tasks.Task`1[[System.__Canon, mscorlib, Version=4.0.0.0, Culture=neutral, PublicKeyToken=b77a5c561934e089]].InnerInvoke()
   at System.Threading.Tasks.Task.Execute()
   at System.Runtime.CompilerServices.TaskAwaiter.ThrowForNonSuccess(System.Threading.Tasks.Task)
   at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(System.Threading.Tasks.Task)
   at Microsoft.VisualStudio.Html.Package.Project.CSS.CssProjectItemData+<StartUpdateClassNameCache>d__43.MoveNext()
   at System.Runtime.CompilerServices.AsyncMethodBuilderCore+<>c.<ThrowAsync>b__6_0(System.Object)
   at System.Windows.Threading.ExceptionWrapper.InternalRealCall(System.Delegate, System.Object, Int32)
   at System.Windows.Threading.ExceptionWrapper.TryCatchWhen(System.Object, System.Delegate, System.Object, Int32, System.Delegate)
   at System.Windows.Threading.DispatcherOperation.InvokeImpl()
   at System.Windows.Threading.DispatcherOperation.InvokeInSecurityContext(System.Object)
   at System.Threading.ExecutionContext.RunInternal(System.Threading.ExecutionContext, System.Threading.ContextCallback, System.Object, Boolean)
   at System.Threading.ExecutionContext.Run(System.Threading.ExecutionContext, System.Threading.ContextCallback, System.Object, Boolean)
   at System.Threading.ExecutionContext.Run(System.Threading.ExecutionContext, System.Threading.ContextCallback, System.Object)
   at MS.Internal.CulturePreservingExecutionContext.Run(MS.Internal.CulturePreservingExecutionContext, System.Threading.ContextCallback, System.Object)
   at System.Windows.Threading.DispatcherOperation.Invoke()
   at System.Windows.Threading.Dispatcher.ProcessQueue()
   at System.Windows.Threading.Dispatcher.WndProcHook(IntPtr, Int32, IntPtr, IntPtr, Boolean ByRef)
   at MS.Win32.HwndWrapper.WndProc(IntPtr, Int32, IntPtr, IntPtr, Boolean ByRef)
   at MS.Win32.HwndSubclass.DispatcherCallbackOperation(System.Object)
   at System.Windows.Threading.ExceptionWrapper.InternalRealCall(System.Delegate, System.Object, Int32)
   at System.Windows.Threading.ExceptionWrapper.TryCatchWhen(System.Object, System.Delegate, System.Object, Int32, System.Delegate)
   at System.Windows.Threading.Dispatcher.LegacyInvokeImpl(System.Windows.Threading.DispatcherPriority, System.TimeSpan, System.Delegate, System.Object, Int32)
   at MS.Win32.HwndSubclass.SubclassWndProc(IntPtr, Int32, IntPtr, IntPtr)

**Comment by Rionmonster** [comment]
Thanks for reporting this @cihanuygun.

I'll take a look at it when I get the chance soon. Do you by chance have any type of example solution or project (or even a Gist) that can easily reproduce this issue? I have Glyphfriend installed and I *know* that I have typed `document.getElementById(...)` at least a few times since installing it.

**Comment by cihanuygun** [comment]
You are welcome @Rionmonster trying to keep good things working well. But I do not know why I could not reproduce this issue on the other (clean new) solutions but only in my private project which includes business logics so I could not share that project.

You can see the javascript code which casues the exception. Take a look @ line 21 https://gist.github.com/cihanuygun/c1d724804574c8fa1505036bb7b89433

Also when I find a chance to debug extension I will try to find issue and report it directly.

**Comment by Shtong** [comment]
Hello

I am having the same issue and have a pretty straightforward way to reproduce it on my setup. Stack trace is the same as the one presented by @cihanuygun 

First off, the environment:

> Microsoft Visual Studio Community 2015
> Version 14.0.25123.00 Update 2
> Microsoft .NET Framework
> Version 4.6.01055
> 
> Installed Version: Community
> 
> Visual Basic 2015   00322-20000-00000-AA241
> Visual C# 2015   00322-20000-00000-AA241
> Visual C++ 2015   00322-20000-00000-AA241
> Visual F# 2015   00322-20000-00000-AA241
> Add New File   3.3
> ASP.NET and Web Tools 2015 (RC1 Update 1)   14.1.20203.0
> ASP.NET Web Frameworks and Tools 2012.2   4.1.41102.0
> ASP.NET Web Frameworks and Tools 2013   5.2.40314.0
> Azure App Service Tools v2.9   14.0.20316.0
> Azure Data Lake Node   1.0
> Azure Data Lake Tools for Visual Studio   2.0.6000.0
> Common Azure Tools   1.7
> GhostDoc   5.1.16036.0
> GitHub.VisualStudio   1.0
> Image Optimizer   3.6.89
> JavaScript Language Service   2.0
> JavaScript Project System   2.0
> Microsoft Azure HDInsight HQL Service   2.0.6000.0
> Microsoft Azure HDInsight Tools for Visual Studio   2.0.6000.0
> Microsoft Azure Mobile Services Tools   1.4
> Microsoft Azure Tools   2.7
> Microsoft Azure Tools   2.9
> NuGet Package Manager   3.4.3
> Open Command Line   2.1.169
> PreEmptive Analytics Visualizer   1.2
> Python Tools for Visual Studio   2.2.40315.00
> Python Tools for Visual Studio - Django Integration   2.2.40315.00
> Python Tools for Visual Studio - Profiling Support   2.2.40315.00
> SQL Server Data Tools   14.0.60311.1
> ToolWindowHostedEditor   1.0
> TypeScript   1.8.30.0
> Visual Rust   0.1
> Web Accessibility Checker   1.1.25
> Web Analyzer   1.7.75
> Web Compiler   1.10.306
> Web Essentials 2015.2   2.0.214

I just installed the extension among all those included in the "Web Extension Pack" https://visualstudiogallery.msdn.microsoft.com/f3b504c6-0095-42f1-a989-51d5fc2a8459 and didn't make the test with only Glyphfriend installed.

Repro steps:
1. Open VS
2. Create a new projet: Visual C# -> Web -> ASP.NET Web Application
3. Choose ASP.NET 5 Templates -> Web Application, with default settings (Individual user accout authentication, no cloud).
4. In the new project, open `wwwroot/js/site.js`
5. In the opened file, type the following:
```javascript
// Write your Javascript code.
function yo() {
    document.getElementById();
}
```
6. While inside the `getElementById` parenthisis, press your double quotes key. Visual Studio crashes there.


Hope this helps

**Comment by Rionmonster** [comment]
@shtong

I really appreciate the repro. I'll check this out as soon as I get a chance and push up a fix for it. Null references are usually :cake to fix.

**Comment by Rionmonster** [comment]
This should be resolved in the latest release 1.5; Closing for now.
